include/dlaf/permutations/general/impl.h

```diff
@@ -139,13 +139,15 @@ void Permutations<B, D, T, C>::call(SizeType i_begin, SizeType i_end, Matrix<con
   SizeType ntiles = i_end - i_begin + 1;
 
   auto sender =
-      ex::when_all(ex::when_all_vector(ut::collectReadTiles(LocalTileIndex(i_begin, 0),
-                                                            LocalTileSize(ntiles, 1), perms)),
-                   ex::when_all_vector(ut::collectReadWriteTiles(LocalTileIndex(i_begin, i_begin),
-                                                                 LocalTileSize(ntiles, ntiles), mat_in)),
-                   ex::when_all_vector(ut::collectReadWriteTiles(LocalTileIndex(i_begin, i_begin),
-                                                                 LocalTileSize(ntiles, ntiles),
-                                                                 mat_out)));
+      ex::when_all(ex::when_all_vector(
+                       matrix::selectRead(perms, common::iterate_range2d(LocalTileIndex(i_begin, 0),
+                                                                         LocalTileSize(ntiles, 1)))),
```

Comment:
Note that this is slightly more verbose when using the generic `selectRead` instead of the old `collectRead(Write)Tiles`. Do you think it makes sense to add an overload of `selectRead(Write)` that directly takes the beginning and end and calls `iterate_range2d`?

Alternatively, the ranges could be separated out into variables. In the two `select` calls below the same range is used.

Any preferences? If none, I'll leave it as it is.

Reply:
I would add an overload which select the full matrix.

Reply:
You mean calls like this, right: https://github.com/msimberg/DLA-Future/blob/79d676defd4bc500082332d095ba041ed59b5a54/include/dlaf/permutations/general/impl.h#L193-L194? How about calling it `select(Read)All` or `select(Read)AllLocal`?

Reply:
Alternatively, to mirror the regular `read` and `readwrite` members, they could be called `readAllLocal` and `readwriteAllLocal`?

Reply:
FYI @rasolca I have not changed anything additional about this (i.e. I have not added additional `select*` overloads). Let me know if you'd still like to see something changed in this PR.

Reply:
Let's move it to another PR.